content/client-lib-development-guide/features.textile

```diff
@@ -332,7 +335,7 @@ h3(#realtime-connection). Connection
 ** @(RTN15c)@ The system's response to a resume request will be one of the following:
 **** @(RTN15c1)@ @CONNECTED@ @ProtocolMessage@ with the same @connectionId@ as the current client, and no @error@. In this case, the server is indicating that the resume succeeded, all channels are still attached, and all backlog messages are available. The client should not change the state of attached channels, and immediately process any queued messages for that channel
 **** @(RTN15c2)@ @CONNECTED@ @ProtocolMessage@ with the same @connectionId@ as the current client, and an @error@. In this case, the server is indicating that the resume succeeded but with a non-fatal error, all channels are still attached, and some backlog messages may be unavailable. The @ErrorInfo@ received should be emitted on the client @Connection@ and the @Connection#errorReason@ should be set. The client should not change the state of attached channels, and immediately process any queued messages for that channel. Any channels that are not resumed in full may receive an @ATTACHED@ @ProtocolMessage@ with an @error@, see "@RTL12@":#RTL12
-**** @(RTN15c3)@ @CONNECTED@ @ProtocolMessage@ with a new @connectionId@, and an error in @error@. In this case, a new connection has been established, the resume has failed, the channels are no longer attached, and the error indicates the resume problem which should be emitted. The client library should immediately detach all channels, fail any queued messages on those channels, and set the @Channel#errorReason@ on each detached @Channel@. Additionally, the internal @msgSerial@ counter is reset so that the first message published to Ably will contain a @msgSerial@ value of @0@
+**** @(RTN15c3)@ @CONNECTED@ @ProtocolMessage@ with a new @connectionId@, and an error in @error@. In this case, a new connection has been established, the resume has failed, the channels are no longer attached, and the error indicates the resume problem which should be emitted. The client library should reattach all channels that are in the @SUSPENDED@ state. For all @ATTACHING@ or @ATTACHED@ channels, the client library should fail any previously queued messages. An @ERROR@ event should additionally be emitted on each @Channel@ object that was automatically re-attached. Finally, the internal @msgSerial@ counter is reset so that the first message published to Ably will contain a @msgSerial@ value of @0@
```

Comment:
> The client library should reattach all channels that are in the @SUSPENDED@ state

If the client attempted a resume, there won't be any channels in the suspended state. Just attached or attaching.

Re the next sentence - in addition to Paddy's suggestion we could be more explicit about the next channel state (ie attaching), eg: "`For all @ATTACHING@ or @ATTACHED@ channels, the client library should fail any previously queued messages, move them to the @ATTACHING@ state, and initiate an attach operation.`"

> can a channel be attached at the point of connection?

The context of this spec item is a resume